### PR TITLE
WIP: Meson: fix syscall sdbs generation

### PR DESCRIPTION
**Pull request by pelijah**
+ handling "symlinks" on Windows
+ replace `gen.sh` with python script for preprocessing syscall sdbs

### CODE REVIEW

File: libr/syscall/d/meson.build [hunk 1]
```diff
@@ -21,29 +21,43 @@ sdb_files = [
...
   py3_exe,
   '-c',
-  'from sys import argv; __import__("os").system("@0@ <%s >%s" % (argv[1], argv[2]))'.format(sdbtmp_gen_path),
+  sdb_pre_script,
   '@INPUT@',
   '@OUTPUT@'
 ]
 
 foreach file : sdb_files
+  infile = run_command(sdb_symlink_cmd + [join_paths(meson.current_source_dir(),'@0@.sdb.txt'.format(file))]).stdout().strip()
```
Comment (ret2libc) [inline libr/syscall/d/meson.build]: why is this necessary? i'm not sure i understand what it does.

File: libr/syscall/d/meson.build [hunk 2]
```diff
@@ -21,29 +21,43 @@ sdb_files = [
...
-sdbtmp_gen_cmd = [
+sdb_pre_script = '''#script
+import re
+import sys
+inf = open(sys.argv[1])
+outf = open(sys.argv[2], 'w')
+for line in inf:
+    if not line.startswith('_') and '=' in line:
+        arr = re.split('=|,', line)
+        print('%s.%s=%s' % (arr[1], arr[2], arr[0]), file=outf)
```
Comment (ret2libc) [inline libr/syscall/d/meson.build]: can you use the same style for everything? either all `print(.., file=outf)` or all `outf.write(...)`?

File: shlr/sdb/meson.build [hunk 3]
```diff
@@ -58,6 +58,26 @@ sdb_exe = executable('sdb', 'src/main.c',
   implicit_include_directories: false
 )
 
+sdb_symlink_script = '''#script
```
Comment (ret2libc) [inline shlr/sdb/meson.build]: this should probably also go in https://github.com/radare/sdb, right @radare ?

File: libr/syscall/d/meson.build (hunk 1 above)
Comment (pelijah) [reply libr/syscall/d/meson.build]: To resolve "symlinks" on Windows. By default git convert symlinks to a text file with a path to the file it points to.

File: shlr/sdb/meson.build (hunk 3 above)
Comment (pelijah) [reply shlr/sdb/meson.build]: I will move it to another place.

**Comment by pelijah** [comment]
Why change title? I'm not going to add new commits.

File: libr/syscall/d/meson.build (hunk 2 above)
Comment (pelijah) [reply libr/syscall/d/meson.build]: Fixed

**Comment by ret2libc** [comment]
@pelijah i didn't see you already fixed the comments ;)
